tests/test_objectives.c

```diff
@@ -1,9 +1,121 @@
 #include <float.h>
 #include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
 #include "objectives.h"
 
 #include <criterion/criterion.h>
 
+#ifndef M_PI
```

Comment:
please implement the objectives in `src/objectives.c`. The `test/test_objectives.c` is only for the unit tests to check the validity of the objective functions. You can then write the unit tests in this file to check you implementation written in `src/objectives.c` and compile/run the tests based on the information provided by Matthäus in the README.

Just as a side note, we try to keep all the testing separate from the implementation because we don't want to bloat our executable once we check for performance. Having code that contains tests compiled into them would be very inefficient and might heavily affect performance as well.